wine64Packages.unstable: 8.17 -> 8.20

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for wine64Packages.unstable is: An Open Source implementation of the Windows API on top of X, OpenGL, and Unix

meta.homepage for wine64Packages.unstable is: https://www.winehq.org/




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 8.18 with grep in /nix/store/i633xkg8gvb11kvq5qmv71nhyi84k1zs-wine64-8.18
- found 8.18 in filename of file in /nix/store/i633xkg8gvb11kvq5qmv71nhyi84k1zs-wine64-8.18

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
27 total rebuild path(s)

27 package rebuild(s)

First fifty rebuilds by attrpath
grapejuice
pipelight
winbox
wine-staging
wine-wayland
wine64Packages.staging
wine64Packages.stagingFull
wine64Packages.unstable
wine64Packages.unstableFull
wine64Packages.waylandFull
winePackages.staging
winePackages.stagingFull
winePackages.unstable
winePackages.unstableFull
winePackages.waylandFull
wineWow64Packages.staging
wineWow64Packages.stagingFull
wineWow64Packages.unstable
wineWow64Packages.unstableFull
wineWow64Packages.waylandFull
wineWowPackages.staging
wineWowPackages.stagingFull
wineWowPackages.unstable
wineWowPackages.unstableFull
wineWowPackages.waylandFull
yabridge
yabridgectl
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/i633xkg8gvb11kvq5qmv71nhyi84k1zs-wine64-8.18 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A wine64Packages.unstable https://github.com/r-ryantm/nixpkgs/archive/d032a901756af612cab35a8b21ade5878a15887c.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/d032a901756af612cab35a8b21ade5878a15887c#wine64Packages.unstable
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/i633xkg8gvb11kvq5qmv71nhyi84k1zs-wine64-8.18
ls -la /nix/store/i633xkg8gvb11kvq5qmv71nhyi84k1zs-wine64-8.18/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

nixpkgs-review took longer than 45m and timed out

---

###### Maintainer pings

cc @avnik @7c6f434c @bendlas @jmc-figueira @reckenrode for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

reckenrode commented:
That staging hash doesn’t look right.

SuperSamus commented:
Probably because it assumes that staging's version is inherited, but it isn't anymore due to #259726.

SuperSamus commented:
Also, #263016 is a duplicate.

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for wine64Packages.unstable is: An Open Source implementation of the Windows API on top of X, OpenGL, and Unix

meta.homepage for wine64Packages.unstable is: https://www.winehq.org/




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 8.20 with grep in /nix/store/qsw1zrrq5wf0l100gs3ijvws8gjnc6aw-wine64-8.20
- found 8.20 in filename of file in /nix/store/qsw1zrrq5wf0l100gs3ijvws8gjnc6aw-wine64-8.20

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
27 total rebuild path(s)

27 package rebuild(s)

First fifty rebuilds by attrpath
grapejuice
pipelight
winbox
wine-staging
wine-wayland
wine64Packages.staging
wine64Packages.stagingFull
wine64Packages.unstable
wine64Packages.unstableFull
wine64Packages.waylandFull
winePackages.staging
winePackages.stagingFull
winePackages.unstable
winePackages.unstableFull
winePackages.waylandFull
wineWow64Packages.staging
wineWow64Packages.stagingFull
wineWow64Packages.unstable
wineWow64Packages.unstableFull
wineWow64Packages.waylandFull
wineWowPackages.staging
wineWowPackages.stagingFull
wineWowPackages.unstable
wineWowPackages.unstableFull
wineWowPackages.waylandFull
yabridge
yabridgectl
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/qsw1zrrq5wf0l100gs3ijvws8gjnc6aw-wine64-8.20 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A wine64Packages.unstable https://github.com/r-ryantm/nixpkgs/archive/674fcd4453dfcbd7e036fcada3ff1e923d26004e.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/674fcd4453dfcbd7e036fcada3ff1e923d26004e#wine64Packages.unstable
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/qsw1zrrq5wf0l100gs3ijvws8gjnc6aw-wine64-8.20
ls -la /nix/store/qsw1zrrq5wf0l100gs3ijvws8gjnc6aw-wine64-8.20/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

nixpkgs-review took longer than 45m and timed out

---

###### Maintainer pings

cc @avnik @7c6f434c @bendlas @jmc-figueira @reckenrode for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

reckenrode commented:
I’ll take a look at this and #263016. I’ve been focused on getting the clang 16 update through staging-next, but it would be nice to have Wine updated before the release and wineWow64 fixed on Darwin.

reckenrode committed:
wine64Packages: 8.17 -> 8.20

https://www.winehq.org/announce/8.18
https://www.winehq.org/announce/8.19
https://www.winehq.org/announce/8.20

reckenrode committed:
wineWow64Packages: fix build on Darwin

reckenrode commented:
The wineWow64Packages fix for Darwin was pretty easy. I have nixpkgs-review running on both Darwin and Linux. Once they’re done, I’ll push an update to this PR with the staging hash and Darwin fixes. (And since staging-next hit master, Darwin should also build again.)

reckenrode commented:
I confirmed locally that it builds on both Darwin and Linux, so this is ready for review.

reckenrode committed:
wine64Packages.minimal: mark broken on Darwin

Darwin requires MinGW to build Wine because clang does not support
`-mabi=ms`, which is required when building without MinGW .

reckenrode commented:
Result of `nixpkgs-review pr 263014` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>grapejuice</li>
    <li>grapejuice.dist</li>
    <li>pipelight</li>
    <li>winbox</li>
    <li>wine-staging (winePackages.stagingFull)</li>
    <li>wine-wayland (winePackages.waylandFull)</li>
    <li>wine64Packages.staging</li>
    <li>wine64Packages.stagingFull</li>
    <li>wine64Packages.unstable</li>
    <li>wine64Packages.unstableFull</li>
    <li>wine64Packages.wayland</li>
    <li>wine64Packages.waylandFull</li>
    <li>winePackages.staging</li>
    <li>winePackages.unstable</li>
    <li>winePackages.unstableFull</li>
    <li>winePackages.wayland</li>
    <li>wineWow64Packages.staging</li>
    <li>wineWow64Packages.stagingFull</li>
    <li>wineWow64Packages.unstable</li>
    <li>wineWow64Packages.unstableFull</li>
    <li>wineWow64Packages.wayland</li>
    <li>wineWow64Packages.waylandFull</li>
    <li>wineWowPackages.staging</li>
    <li>wineWowPackages.stagingFull</li>
    <li>wineWowPackages.unstable</li>
    <li>wineWowPackages.unstableFull</li>
    <li>wineWowPackages.wayland</li>
    <li>wineWowPackages.waylandFull</li>
    <li>yabridge</li>
    <li>yabridgectl</li>
  </ul>
</details>

reckenrode commented:
Result of `nixpkgs-review pr 263014` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>wine64Packages.minimal</li>
    <li>wineWow64Packages.minimal</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>wine64 (wine64Packages.stableFull)</li>
    <li>wine64Packages.base (wine64Packages.stable)</li>
    <li>wine64Packages.staging</li>
    <li>wine64Packages.stagingFull</li>
    <li>wine64Packages.unstable</li>
    <li>wine64Packages.unstableFull</li>
    <li>wineWow64Packages.base (wineWow64Packages.stable)</li>
    <li>wineWow64Packages.full (wineWow64Packages.stableFull)</li>
    <li>wineWow64Packages.staging</li>
    <li>wineWow64Packages.stagingFull</li>
    <li>wineWow64Packages.unstable</li>
    <li>wineWow64Packages.unstableFull</li>
  </ul>
</details>

reckenrode commented:
I also added a commit to mark the minimal packages broken on Darwin since they need MinGW. It’s a waste of time trying to build them, and we’re in ZHF anyway.
